sdk/rust/oak_utils/src/lib.rs

```diff
@@ -14,159 +14,165 @@
-// that are not present in `temp_path`.
-fn update_files(temp_path: &Path, path: &Path) -> io::Result<()> {
-    let files = get_files(path);
-    let temp_files = get_files(temp_path);
-    get_file_diffs(&files, &temp_files)
-        .iter()
-        .try_for_each(|diff| {
-            let filepath = path.join(&diff.filename);
-            match diff.action() {
-                FileAction::Ignore => Ok(()),
-                FileAction::Remove => fs::remove_file(filepath),
-                FileAction::Update { new_content } => {
-                    // `File::create` truncates an existing file.
-                    let mut file = fs::File::create(filepath).unwrap();
-                    file.write_all(new_content.as_bytes())
-                }
+            #[allow(dead_code)]
+            pub struct #client_name(pub #oak_package::grpc::client::Client);
+
+            #[allow(dead_code)]
+            impl #client_name {
+                #(#client_methods)*
             }
-        })
+        };
+        out.push_str("\n");
+        out.push_str(&format!("{}", gen));
+        out.push_str("\n");
+    }
 }
 
-// Returns a hash set of file differences that represent old and new contents.
-fn get_file_diffs(
-    old_files: &HashMap<String, String>,
-    new_files: &HashMap<String, String>,
-) -> HashSet<FileDiff> {
-    let file_set = old_files
-        .keys()
-        .chain(new_files.keys())
-        .cloned()
-        .collect::<HashSet<String>>();
-    file_set
-        .iter()
-        .map(|filename| FileDiff {
-            filename: filename.to_string(),
-            old_content: old_files.get(filename).cloned(),
-            new_content: new_files.get(filename).cloned(),
-        })
-        .collect()
-}
+pub fn compile_protos<P>(inputs: &[P], includes: &[P])
+where
+    P: AsRef<std::path::Path>,
+{
+    for input in inputs {
+        // Tell cargo to rerun this build script if the proto file has changed.
+        // https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorerun-if-changedpath
+        println!("cargo:rerun-if-changed={}", input.as_ref().display());
+    }
 
-// Traverses `dir` and produces a `HashMap` of filenames and their contents.
-fn get_files(dir: &Path) -> HashMap<String, String> {
-    walkdir::WalkDir::new(dir)
-        .into_iter()
-        .filter_map(|entry| entry.ok())
-        .filter(|entry| entry.path().is_file())
-        .map(|entry| {
-            let path = entry.into_path();
-            let content = fs::read_to_string(&path).expect("Read error");
-            let filename = path
-                .file_name()
-                .expect("Filename error")
-                .to_os_string()
-                .into_string()
-                .expect("OsString error");
-            (filename, content)
-        })
-        .collect()
+    prost_build::Config::new()
+        .service_generator(Box::new(OakServiceGenerator))
+        .type_attribute(".oak.label", "#[derive(Eq, Hash)]")
```

Comment:
(pre-existing) comment for why oak.label is special?

Reply:
Done.